lib/eco/export/jruby.py

```diff
@@ -381,3 +389,63 @@ def _profile(self):
                              num_calls))
                 self._annotate_text(method.source.line_start, method.name,
                                     dmsg, JRubyMorphismMsg)
+                # Find and annotate method calls.
+                for version in method.versions:
+                    for callsite_version in version.called_from:
+                        if method.is_mega:
+                            cmsg = ('Call to %s defined on line %g (megamorphic).' %
+                                    (method.name, method.source.line_start))
+                        else:
+                            cmsg = ('Call to %s defined on line %g.\n' %
+                                    (method.name, method.source.line_start))
+                        cmsg += '\n(line numbers may be inaccurate in polyglot programs)'
+                        self._annotate_text(callsite_version.callsite.line,
+                                            method.name, cmsg, JRubyMorphismMsg)
+                        if callsite_version.callsite.line > 0:
+                            self._annotate_text(callsite_version.callsite.line,
+                                                method.name,
+                                                { method.name : method.is_mega },
+                                                JRubyMorphismLine)
+                        num_calls += 1
+
+    def _annotate_text(self, lineno, text, annotation, klass):
+        """Annotate a node on a given line with a given symbol name.
+        """
+        if lineno < 0:
+            return
+        # Attempt to find 'text' on 'lineno'.
```

Comment:
It isn't ideal, but I'm trying to avoid searching the whole tree, where possible.